modules/dxp/apps/commerce/commerce-shop-by-diagram/commerce-shop-by-diagram-web/src/main/resources/META-INF/resources/js/ImagePins.js

```diff
@@ -0,0 +1,214 @@
+import {
+	moveDown,
+	moveLeft,
+	moveRight,
+	moveUp,
+	zoomIn,
+	zoomOut,
+} from './NavigationsUtils';
+import ZoomController from './ZoomController';
+
+const ImagePins = ({
+	changedScale,
+	enablePanZoom,
+	execZoomIn,
+	image,
+	imageSettings,
+	navigationController,
+	resetZoom,
+	selectedOption,
+	setChangedScale,
+	setResetZoom,
+	setSelectedOption,
+	setZoomInHandler,
+	setZoomOutHandler,
+	spritemap,
+	zoomController,
+	zoomInHandler,
+	zoomOutHandler,
+}) => {
+	const handlers = useRef();
+	const container = useRef();
+	const panZoom = useRef();
+
+	const svg = useRef(null);
+
+	useLayoutEffect(() => {
+		container.current = select('g#container');
+		panZoom.current = zoom()
+			.scaleExtent([0.5, 40])
+			.on('zoom', () => {
+				container.current.attr('transform', event.transform);
+			});
+
+		if (enablePanZoom) {
+			container.current.call(panZoom.current);
+		}
+
+		if (resetZoom) {
+			setResetZoom(false);
+			container.current
+				.transition()
+				.duration(700)
+				.call(
+					panZoom.current.transform,
+					zoomIdentity,
+					zoomTransform(container.current.node()).invert([
+						imageSettings.width,
+						imageSettings.height,
+					])
+				);
+
+			setSelectedOption(1);
+		}
+
+		if (changedScale) {
+			setChangedScale(false);
+			const imageInfos = container.current.node().getBBox();
+			container.current
+				.transition()
+				.duration(700)
+				.attr(
+					'transform',
+					`translate(${imageInfos.width / 2.0},${
+						imageInfos.height / 2.0
+					}) scale(${selectedOption}) translate(-${
+						imageInfos.width / 2.0
+					},-${imageInfos.height / 2.0})`
+				);
+		}
+
+		if (execZoomIn) {
+			handlers.current?.zoomIn();
+		}
+
+		if (zoomOutHandler) {
+			setZoomOutHandler(false);
+			zoomOut(container.current, panZoom.current);
+		}
+
+		if (zoomInHandler) {
+			setZoomInHandler(false);
+			zoomIn(container.current, panZoom.current);
+		}
+
+		handlers.current = {
+			moveDown: () => moveDown(container.current, navigationController),
+			moveLeft: () => moveLeft(container.current, navigationController),
+			moveRight: () => moveRight(container.current, navigationController),
+			moveUp: () => moveUp(container.current, navigationController),
+			zoomIn: () => zoomIn(container.current, panZoom.current),
+			zoomOut: () => zoomOut(container.current, panZoom.current),
+		};
+	}, [
+		resetZoom,
+		selectedOption,
+		setResetZoom,
+		changedScale,
+		zoomOutHandler,
+		zoomInHandler,
+		enablePanZoom,
+		execZoomIn,
+		imageSettings,
+		navigationController,
+		setChangedScale,
+		setSelectedOption,
+		setZoomInHandler,
+		setZoomOutHandler,
+	]);
+
+	const diagramStyle = {
+		height: `${imageSettings.height}`,
+		width: `${imageSettings.width}`,
+	};
+
+	return (
+		<div className="diagram-pins-container" style={diagramStyle}>
+			<svg
+				height={imageSettings.height}
+				ref={svg}
+				width={imageSettings.width}
+			>
+				<g
+					data-testid="container"
+					id="container"
```

Comment:
We should namespace this id with the namespace of the portlet so that it doesn't collide with any other id's on the page